### ISSUE TITLE
Client/server status mismatch.

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-web#8246]
I loaded Riot and instead of the room name I saw 'Random Participant and 3788 Others'

**Comment by aaronraimist** [comment]
I've never had this problem but this is probably the most commonly asked about issue. Often in busy rooms people will ask what room they are in, usually thinking that it is either some kind of spam room they were auto joined to or something. Most of the time asking them to clear cache and reload fixes it.

**Comment by lukebarnard1** [comment]
I joined with a new account and got this on two of three rooms I joined to begin with. Will rageshake!

**Comment by aaronraimist** [comment]
Just got this on my matrix.org account, rageshake sent.

**Comment by aaronraimist** [comment]
I've added a $200 bounty to encourage a community member to work on this issue. https://freedomsponsors.org/issue/850/missing-room-name-fixed-by-clear-cache?alert=SPONSOR#sponsors

Even don't experience it regularly myself, it is one of the most common issues I see people complain about and **it leads people to think that Riot is extremely broken**. Most think Riot auto joined them to some spam room or something and it is extremely off-putting for a non technical user. The bounty expires on 2020-01-01.

If a New Vector employee solves the issue while at work, I will reallocate this bounty to another issue. If a core team member wants to solve this with their community hat on then they can choose to take the bounty.

**Comment by loru** [comment]
Same issue for me
![Screen Shot 2019-04-17 at 6 25 54 PM](https://user-images.githubusercontent.com/19887885/56304870-55184a80-613f-11e9-9cec-570a816dd1eb.png)

**Comment by sunshineo** [comment]
Hi @aaronraimist, I made a [Github bot](https://github.com/apps/issue-reward-beta) and a [website](https://www.issuereward.com/) similar to the https://freedomsponsors.org/ you are using. The Github bot can automatically comment on the issue and tag people when anyone gives a reward, claim the rewards, and approve a reward. Automatically label issues with rewards so you know what is really important for your users.

Could you try it on this repo? It's in beta now, I'm looking for people to give it a try.

**Comment by DrugsMaster** [comment]
This bug happens not only in riot, but in fractal too. After AFKing for 8-10 hours the official Matrix HQ room (#matrix:matrix.org) was named "bob and Friends" to me. I also couldn't write messages in to chat. Restarting app helped me.

**Comment by jplatte** [comment]
Still happening. Can I do anything to help debug this?

**Comment by jplatte** [comment]
Also happening [in element-android](https://github.com/vector-im/element-android/issues/1873) (to me too, with the same room on both actually). I wonder whether this is really a client-side issue.

**Comment by jplatte** [comment]
By the way, I've only seen this happen with v1 rooms. Then again, this is only the second time I've seen it.

**Comment by joepie91** [comment]
Any movement on this? It's starting to become a daily question in HQ.

**Comment by jplatte** [comment]
This _doesn't_ happen in v1 rooms only, though it seems like when this happens it can also lead to a room falsely being shown as being v1. Just had this happen to the Conduit room, _and it persisted even through clearing the cache_!

**Comment by TheGreatestJannet** [comment]
I get this issue extremely often on both Element Desktop as well as Element-iOS. Clearing the cache fixes it every time but it is quite annoying!

**Comment by SimonBrandner** [comment]
Related https://github.com/vector-im/element-web/issues/20064

**Comment by timokoesters** [comment]
> This _doesn't_ happen in v1 rooms only, though it seems like when this happens it can also lead to a room falsely being shown as being v1. Just had this happen to the Conduit room, _and it persisted even through clearing the cache_!

This might have been a state reset and not this bug.

**Comment by jplatte** [comment]
Hm yeah, though a state reset shouldn't result in a room being shown as a different room version..

**Comment by timokoesters** [comment]
Maybe the client couldn't find the create event and defaulted to v1

**Comment by weedoe** [comment]
Having the same issue, and does not fix by clearing the cache. The problem occurs on all clients that I'm signed into. iOS, web and desktop are affected.

**Comment by aaronraimist** [comment]
@weedoe that's probably a different issue. Is it a version 1 room?

**Comment by weedoe** [comment]
@aaronraimist Version 9

**Comment by aaronraimist** [comment]
@weedoe can other people in the room see the name? What happens if they clear cache?

**Comment by weedoe** [comment]
Everyone else can see the room name, unsure about other accounts on same homeserver

**Comment by timmc** [comment]
Yeah, currently seeing this too. In my newer element-web session (in-browser), the room has its proper name, description, avatar, and published address and is showing as room version 6. In my older session (on the desktop), all of those are reset and the room version is showing as 1. Element version is 1.11.16 in both cases, and server is Dendrite v.0.10.8. I *am* able to post new messages in the room.

Please let me know if there is anything I can do to help gather more info or debug this. (I'd rather not submit the entire rageshake tarball unreviewed, but I can upload extracts of it, search for IDs, etc.)

**Comment by xPMo** [comment]
Just want to add a client-side datapoint. I'm in reasonably-busy public room (608 members), and got this bug in Element. `/devtools` > Explore Room State:

![A screenshot of Developer tools in Element, with only m.room.member, im.vector.modular.widgets, and m.room.power_levels](https://github.com/vector-im/element-web/assets/16061366/090bb921-8128-4840-bd96-70b77c1cdce5)

Meanwhile, after authenticating by [another method](https://github.com/fabianonline/matrix.sh) and pulling the state directly:

```sh
❯ . ~/.matrix.sh; curl <SNIP> "$MATRIX_HOMESERVER/_matrix/client/v3/rooms/\!teHukPkJDFppjPjGQg:matrix.org/state" | jq 'map(.type) | unique'
[
  "im.vector.modular.widgets",
  "m.room.avatar",
  "m.room.canonical_alias",
  "m.room.create",
  "m.room.history_visibility",
  "m.room.join_rules",
  "m.room.member",
  "m.room.name",
  "m.room.power_levels",
  "m.room.topic",
  "m.space.parent"
]
```
I can confirm that Element is falling back to v1 because of the missing create event.
